### PR TITLE
Spell Check and Small updates to Devcontainer

### PR DESCRIPTION
**Pull request by chrheg**
## PR Type

- [ ] Adding new question(s)
> **Warning**: We do not support the inclusion of questions directly copied from official GitHub certification exams. Please only submit original questions and content that you have created.`
- [x] Other content changes (updating questions, answers, explanations or study resources)
- [x] Code changes (non-content)
- [ ] Documentation changes
- [ ] Other

## What's new?

- Optimized devcontainer for shallow clone, increasing codespace startup time.
- Integrated Node into the devcontainer image.
- Included cSpell into the devcontainer image.
- Implemented workflow to spell-check markdown files in the content directory.
- Performed spell-check on all existing files and made necessary updates.

### CODE REVIEW

File: .devcontainer/devcontainer.json
Comment (FidelusAleksander) [inline .devcontainer/devcontainer.json:1]: 1) Let's not add Node and cspell to devcontainers. Spell checking as part of Pull request checks is enough. Seriously doubt people will be running cspell command prior to pushing. They will see it  as PR check from github action

Comment (FidelusAleksander) [reply .devcontainer/devcontainer.json:1]: Shallow cloning is nice, good find

**Review by FidelusAleksander — CHANGES_REQUESTED** [review]
Hey @chrheg 

Thanks for adding this! It's a solid addition. Left few small comments but overall it's good

File: .github/workflows/pr-checks.yml [hunk 1]
```diff
@@ -20,3 +20,12 @@ jobs:
...
       - uses: pre-commit/action@v3.0.0
+  spell-check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+      - name: Install cspell
+        run: npm install -g cspell
+      - name: Run Spell Checker
+        run: find ./content -name "*.md" | xargs cspell
```
Comment (FidelusAleksander) [inline .github/workflows/pr-checks.yml:31]: Could use this action instead, but this is fine aswell. I might replace it later with the action, unless you want to do that
https://github.com/streetsidesoftware/cspell-action

File: cSpell.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
+{
+    "version": "0.2",
+    "language": "en",
+    "words": [
+        "autobuild",
+        "automations",
+        "codebases",
+        "CODEOWNERS",
+        "CODEOWNERS",
```
[the remark this answers is not in the document]
Comment (chrheg) [reply cSpell.json:9]: Sorted out, was a bit quick on this it seems :(

File: .github/workflows/pr-checks.yml (hunk 1 above)
Comment (chrheg) [reply .github/workflows/pr-checks.yml]: That one is even better, update the code to use it now.

Comment (FidelusAleksander) [reply .github/workflows/pr-checks.yml]: @chrheg Seems like it has issues reading the config file
https://github.com/FidelusAleksander/ghcertified/actions/runs/7596864581/job/20691056083?pr=159

File: .github/workflows/pr-checks.yml [hunk 3]
```diff
@@ -20,3 +20,14 @@ jobs:
       - uses: actions/checkout@v4
       - uses: actions/setup-python@v5
       - uses: pre-commit/action@v3.0.0
+  spell-check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
```
Comment (FidelusAleksander) [inline .github/workflows/pr-checks.yml:27]: `setup-node` is not needed, already done by cspell-action

**Review by FidelusAleksander — APPROVED** [review]
LGTM!

File: .github/workflows/pr-checks.yml (hunk 1 above)
Comment (chrheg) [reply .github/workflows/pr-checks.yml]: Had to add the name of the config file to the action, looks like its running now.

**Pull request merged**